### DISCUSSION TITLE
remote machine learning address

### DISCUSSION THREAD

**Discussion by summX** [immich-app/immich#9897] [General]
Having an existential crisis here: unraid docker, set machine learning host to IP of that host, set machine learning in immich admin panel to 0.0.0.0:3003. Worked this morning. Then I was forced to redo my unraid setup and now I´m stuck with "invalid address" in the log and can´t connect to the machine learning host. 
I am not certain why immich needs this kind of "double" entry of values and one does not know what exactly to place where.

**Comment by bo0tzz** [comment]
0.0.0.0 is not a valid IP address to reach something on - it refers to "every interface" when binding a service to a port on a system. You need to figure out the actual IP of the remote host.

**Reply by summX** [reply]
> 0.0.0.0 is not a valid IP address to reach something on - it refers to "every interface" when binding a service to a port on a system. You need to figure out the actual IP of the remote host.

strangely it was working. I had the IP address of my remote learning machine in docker settings and 0.0.0.0:3003 set in immich ui admin panel. Anyway. So I need to set docker settings to remote host IP AND immich admin panel? this still does not work as I get ERROR: invalid address ('IP', 3003) and it won´t connect. 

I can test with telnet ip 3003 and it works. It was working before.

**Reply by summX** [reply]
It now works with 0.0.0.00 as MACHINE_LEARNING_HOST and IP of host in admin panel.

**Reply by dbatkinsmith** [reply]
Does this still work for you, or has it changed with the 1.106.x updates? I'm getting invalid address errors no matter what I try.

**Reply by summX** [reply]
> Does this still work for you, or has it changed with the 1.106.x updates? I'm getting invalid address errors no matter what I try.

So far it does. I find those settings confusing nonetheless.
